pkg/neorpc/rpcevent/filter.go

```diff
@@ -66,7 +67,28 @@ func Matches(f Comparator, r Container) bool {
 		notification := r.EventPayload().(*state.ContainedNotificationEvent)
 		hashOk := filt.Contract == nil || notification.ScriptHash.Equals(*filt.Contract)
 		nameOk := filt.Name == nil || notification.Name == *filt.Name
-		return hashOk && nameOk
+		parametersOk := true
+		if len(filt.Parameters) > 0 {
+			stackItems := notification.Item.Value().([]stackitem.Item)
+			parameters, err := filt.ParametersSI()
+			if err != nil {
+				return false
+			}
+			for i, p := range parameters {
+				if p.Type() == stackitem.AnyT && p.Value() == nil {
+					continue
+				}
+				if i >= len(stackItems) {
+					parametersOk = false
+					break
+				}
+				if p.Type() != stackItems[i].Type() || !p.Equals(stackItems[i]) {
```

Comment:
Remove type check, Equals will handle it.

Reply:
well, i am lost then or do not understand smth. i have added it to follow your https://github.com/nspcc-dev/neo-go/pull/3689#discussion_r1851959957. it was not here originally

Reply:
> i have added it to follow your https://github.com/nspcc-dev/neo-go/pull/3689#discussion_r1851959957.

But I was talking about https://github.com/nspcc-dev/neo-go/blob/990634a43af5be0609270b2fce6b103f92686998/pkg/smartcontract/param_type.go#L193

The link is attached in the conversation above.

Reply:
ok, let's start over. we have caching now, all the parameters are converted at a time so we always compare the same types (`stackitem`) so we don't need `Match` at all right?

Reply:
Yes, with caching no `Match` required. As long as `p.Type() != stackItems[i].Type()` check.